### ISSUE TITLE
Left margin outline pane

### ISSUE THREAD

**Issue by DutchPete** [hydescarf/Obsidian-Theme-Mado-Miniflow#5]
The left margin is relatively wide, how can I reduce it?
I have tried with
```css
outline .tree-item-inner {
  margin-left: -50px;
}
```
That works but the reduction is insufficient. But if I go to e.g. -60px the text goes behind the white.

**Comment by hydescarf** [comment]
Is this particular just for "outline" plugin, or the whole sidebar?
This might probably not be included in the official fixes but here's a workaround:
```css
.outline .tree-item-self{
    padding-inline-start:0;
}
```
That basically turns off the padding-left.

But if you want it like, for the whole sidebar to be clipped to the note pane, the note pane has a rather wide padding on the sides, I'm currently looking for solution for this part and it will be included in the next update.
As for the workaround, try this:
```css
.workspace>.workspace-split.mod-vertical{
    padding:33px 0;
}
```

**Comment by DutchPete** [comment]
Thanks for your quick reply, and for a nice theme!

I have the core plug-in “Outline” enabled, so you get an outline in its own pane in the right sidebar.

As for turning off the padding on the left, that does not work.
The code with `.workspace>.workspace-split.mod-vertical` does not work, and adds an extra empty strip at the bottom of the note.

Just to clarify what I am talking about, see screenshot below.

![image](https://user-images.githubusercontent.com/10794265/183279687-9535d5ef-0013-443d-b4d9-f7d6bc12a355.png)

The white spaced margin on the left is too wide for me, but reducing the `margin-left` to e.g -60px pulls the text more to the left, as it should, but “behind” that white space.

**Comment by hydescarf** [comment]
My mistake, I should've test this on a snippet file instead of on the app directly.
Please try the following again and let me know if it works!
```css
.workspace-split.mod-right-split .view-content .outline .tree-item-self{
    padding-inline-start:0;
}
.workspace>.workspace-split.mod-vertical{
    padding:33px 0 0;
}
```
The first element is for the `.outline` plugin, it just needs a few more classes name before to strengthen it. Or you can just use a `!important`:
```css
.outline .tree-item-self{
    padding-inline-start:0!important;
}
```

The second one is for the main note pane, I forgot to retain the `0px` for the `padding-bottom`.

**Comment by DutchPete** [comment]
```css
.workspace-split.mod-right-split .view-content .outline .tree-item-self{
    padding-inline-start:0;
}
```
Does not work, even with `!important` added.

I don't need to change anything to the main note pane.

```css
.outline .tree-item-self{
    padding-inline-start:0!important;
}
```
That does move it to the left, but like with the `margin-left` rule, it goes “behind” the white:

![image](https://user-images.githubusercontent.com/10794265/183282381-f596a345-c0d1-47a8-96d7-e1094323918a.png)

**Comment by hydescarf** [comment]
The main pane note has additional 23px of padding that forms the empty space to the right, as since you needed to put the margin at negative 50px, I assumed you want almost 0px spaces in between the main pane note and the sidebar, thus the suggestion.

As I'm not able to reproduce that on my side, I'd like to confirm a few things before continuing:

1. Are you using the theme downloaded from the store? Or was this a direct install from Github into the snippet folder?
2. Have you tried disabling/re-enabling the themes/plugins? I've got an issue from someone before where part of the theme were still using the default styles, where reinstalling fixed it.
3. Are there any other CSS changes you've made prior to this (both related and unrelated)? Something might be cutting off the edges of that sidebar. The ordering of the snippet files plays a role, if there is a file that styles the outline named eg. "outline.css" and you named the fixes eg. "mado.css", "mado.css" will be executed first, and then "outline.css" will override the style later.
4. Can I have a bigger picture of the screenshot? The black border beside is intriguing. Since `.outline .tree-item-self` works but not the other, I'm assuming this part `.workspace-split.mod-right-split` is not working.

**Comment by DutchPete** [comment]
To answer your points:

1. I installed the them from the store
2. One can only enable 1 theme at the time so I don't see how I can disable the ohter ones in my themes folder. I disabled all community plug-ins, but there is no difference
3. I have a number of snippets active, so I disabled them 1 by 1, but none affects the outline pane. I have 1 snippet that affects the outline pane, so I disabled it and added your proposed code in a separate snippet. The effect is the same os I had before: the outline headings get pulled behind the white space.
4. The black border is nothing intriguing when you know what it is: my Obsidian window does not cover my screen completely, so that black strip is just the Github window below, which I have set to dark (= black) mode.

BTW, I always have the outline pane dragged over to the left sidebar.

**Comment by hydescarf** [comment]
Thank you for the answers, I think I see what was happening *on my side* instead.
The problem was that I was checking with a file that has no children headers after H1, which results in what I thought was a normal visual appearance. I'm very sorry for the troubles. ><

The solution to this is however, the same as before.
Take a look at this picture:
![image](https://user-images.githubusercontent.com/56707567/183286199-5d33c1ed-4ae0-4f11-aed7-f1061547142f.png)
The blue line indicates the drag-handle that resizes the sidebars when you hover at them.
Going with `0px` from `23px` on the padding of `outline` will results in the cut as appear below:
(Although, the cut is only on the expanding arrow, I still cannot replicate the cut on the text)
![image](https://user-images.githubusercontent.com/56707567/183286269-56798a59-a078-4e2a-9575-8091869b0ccd.png)
Basically, we are already at our limitation as we cannot shrink any more on the sidebar.

So, the only solution is to actually shrink the padding of the main note pane:
![image](https://user-images.githubusercontent.com/56707567/183286477-e31b2560-7f3a-4975-8385-bc077dc47b7a.png)
However, you mentioned that the following code didn't work before
```css
.workspace>.workspace-split.mod-vertical{
    padding:33px 0 0;
}
```
Can you try this again, and this time with the other snippets being disabled? (or try it with `!important`)

**Comment by DutchPete** [comment]
```css
.workspace>.workspace-split.mod-vertical{
    padding:33px 0 0;
}
```

reduces the left margin of the note. However, that may be useful if the outline pane is in the sidebar on the right hand side. As I said in my previous comment, I have my outline pane on the left hand side so this code does not help me.

So far, with all the themes I have tried (always with the outline pane in the left hand sidebar) I have always been able to move the outline content to the left of the screen as far as I wanted without it disappearing behind a white zone, when I use

```css
.outline .tree-item-inner {
  margin-left: whatever;
}
```

**Comment by hydescarf** [comment]
I'm sorry, I have misunderstood your comments a lot, so your issue was that you're trying by `margin-left` to reduce the space on the left on outline that is on `left-sidebar`?
That will be a totally a different issue and is related to issue https://github.com/hydescarf/Obsidian-Theme-Mado-Miniflow/issues/4 which I plan to include in the future update.

Though, to have a workaround for now, try this instead:
```css
.mod-left-split .workspace-leaf-content{
	max-width:initial;
}
```
That will reset the `max-width` to full-width, and will also affect the whole `left-sidebar`.
In this case, the theme is aligning the content through `flex` property, so `margin-left` is actually absent and not being usable.
You can use `max-width` to define the size, or use `padding-left` as opposed to `margin-left` in the code above.

Very sorry for the much troubles!

**Comment by DutchPete** [comment]
YES !!!! That works. Many thanks, and no troubles caused, I am happy to use your theme!
Merci beaucoup. 👍

**Comment by hydescarf** [comment]
Glad it all work!

Actually, we'll leave this open as the main issue still exists at the moment.
Will close once I found a way to update them.

**Comment by hydescarf** [comment]
On a second thought, since there's a similar thread outside, we'll focus on that thread instead. Closing this thread for now!

**Comment by DutchPete** [comment]
Fine with me. Your “DutchPete-customized-workaround” works perfectly, I have put it in its own snippet.
Many thanks for your help and patience!
